### ISSUE TITLE
Documentation about style enforcement

### ISSUE THREAD

**Issue by kbarnhart** [landlab/landlab#846]
@mcflugen I took a stab at writing information about style enforcement [here](https://github.com/landlab/landlab/wiki/Style-conventions#style-enforcement).

Please edit it as you see fit (and then close this issue).

**Comment by kbarnhart** [comment]
Two things that should probably be added.

1) A recommendation on the best way to remove trailing whitespace. For me, `black` and `isort` don't do this. I need to use atom to save without trailing whitespace.

2) A statement about how to handle local variable assignment. Usually I use 
```python
_ = my_func(args, *kwargs)
```
but this yields
`F841 local variable '_' is assigned to but never used`

**Comment by kbarnhart** [comment]
thanks @margauxmouchene! and thank you for telling me about that Spyder option (I've always just opened everything in atom at the end of work to save without whitespace). 🤸‍♀️ 👍 🎉

**Comment by mcflugen** [comment]
@kbarnhart Thank you for getting this started! 👏

For trailing whitespace I usually just remove it by hand as it's pretty rare. If for some reason there's a bunch of it, I do,
```bash
$ sed -i '' -e's/[[:space:]]*$//' <file-name>
```

If you're not going to use `_` why use `_ = my_func(args, *kwargs)` instead of just `my_func(args, *kwargs)`?

**Comment by kbarnhart** [comment]
@mcflugen, I usually use `_` when the function returns something. 
I'm happy to not use anything at all, though.

**Comment by SiccarPoint** [comment]
"Space after all commas" seems most consistent to me, and I'm fairly sure is PEP8 - which remains our declared style.

**Comment by kbarnhart** [comment]
I agree with @SiccarPoint.

Just to make clear where I found this issue: Lines 3251 and 3253 of raster.py

black wants:
```python
neighbors[:,] = self.active_adjacent_nodes_at_node[ids]
diagonals[:,] = self.diagonal_adjacent_nodes_at_node[ids]
```

PEP8 wants:
```python
neighbors[:, ] = self.active_adjacent_nodes_at_node[ids]
diagonals[:, ] = self.diagonal_adjacent_nodes_at_node[ids]
```

**Comment by mcflugen** [comment]
@kbarnhart, @SiccarPoint I agree with both of you. For this case, I've just been going in by hand after running *black* and changing it to something that *flake* won't complain about. I think the best solution might be to change those lines to something without the comma. I'm not really sure how those lines are different from,
```python
neighbors[:] = self.active_adjacent_nodes_at_node[ids]
diagonals[:] = self.diagonal_adjacent_nodes_at_node[ids]
```

**Comment by SiccarPoint** [comment]
Why can't we leave PEP8 as our style...?

**Comment by mcflugen** [comment]
We totally should. I was just saying that having that comma there was confusing.

Looking more closely at [PEP8](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements), I actually think the space should *not* be there (but it's not totally clear).

> Between a trailing comma and a following close parenthesis.
>
> Yes: foo = (0,)
> No:  bar = (0, )

**Comment by SiccarPoint** [comment]
You're right on the lack of clarity. FWIW, my PEP8 linter in Atom enforces the space.

**Comment by kbarnhart** [comment]
@mcflugen there has been some confusion about this (and how it relates to the linting errors in #902).

So we should finalize this and close the issue.
